satellite/mailservice/service.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package mailservice
+
+import (
+	"bytes"
+	"context"
+	htmltemplate "html/template"
+	"path/filepath"
+
+	"go.uber.org/zap"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/post"
+)
+
+// Config defines values needed by mailservice service
+type Config struct {
+	SMTPServerAddress string `help:"smtp server address" default:""`
+	TemplatePath      string `help:"path to email templates source" default:""`
+	From              string `help:"sender email address" default:""`
+	AuthType          string `help:"smtp authentication type" default:"simulate"`
+	PlainLogin        string `help:"plain auth user login" default:""`
+	PlainPassword     string `help:"plain auth user password" default:""`
+	RefreshToken      string `help:"refresh token used to retrieve new access token" default:""`
+	ClientID          string `help:"oauth2 app's client id" default:""`
+	ClientSecret      string `help:"oauth2 app's client secret" default:""`
+	TokenURI          string `help:"uri which is used when retrieving new access token" default:""`
+}
+
+var (
+	mon = monkit.Package()
+)
+
+// Sender is
+type Sender interface {
+	SendEmail(msg *post.Message) error
+	FromAddress() post.Address
+}
+
+// Message defines mailservice template-backed message for SendRendered method
+type Message interface {
+	Template() string
+	Subject() string
+}
+
+// Service sends template-backed email messages through SMTP
+type Service struct {
+	log    *zap.Logger
+	sender Sender
+
+	html *htmltemplate.Template
+	// TODO(yar): prepare plain text version
+	//text *texttemplate.Template
+}
+
+// New creates new service
+func New(log *zap.Logger, sender Sender, templatePath string) (*Service, error) {
+	var err error
+	service := &Service{log: log, sender: sender}
+
+	// TODO(yar): prepare plain text version
```

Comment:
Can we include this, instead of leaving it as TODO?

Reply:
Will ad as soon as I have content